### DISCUSSION TITLE
error when use bignum in query

### DISCUSSION THREAD

**Discussion by oRIOnTx** [mongodb/mongoid#5784] [Q&A]
I have problem when i try user bignum in my query. I have next error: /usr/local/bundle/ruby/3.0.0/gems/bson-4.15.0/lib/bson/hash.rb:43:in `put_hash': bignum too big to convert into `long long' (RangeError)

i have next code for example: 
ip_code_str = 58569104840549217479400131697037017089
::Mongo::Country.where(:range_from.lte => ip_code_str, :range_to.gte => ip_code_str).first

i'm trying covert number to string, but nothing changed

**Comment by alexbevi** [comment]
Hey @oRIOnTx,

Based on your variable name of `ip_code_str` did you mean to represent that value as a string instead of a number? If not, could you share your model so we can attempt to reproduce this to better understand the issue?

**Reply by oRIOnTx** [reply]
i have next code variants: 
      ip_code = 58569104840549217479400131697037017089
      anonymous = ::Mongo::Proxy.where(:range_from.lte => ip_code).and(:range_to.gte => ip_code).first
&

  ip_code = 58569104840549217479400131697037017089
      anonymous = ::Mongo::Proxy.where(:range_from.lte => "#{ip_code}").and(:range_to.gte => "#{ip_code}").first

&

  ip_code = 58569104840549217479400131697037017089
      anonymous = ::Mongo::Proxy.where(:range_from.lte => ip_code.to_s).and(:range_to.gte => ip_code.to_s).first

but nothing helped. 
sample query what i want to execute by mongoid
db.getSiblingDB("my_db").getCollection("mongo_countries").find({
    'range_from': { '$lte': "58569104840549217479400131697037017089" },
    'range_to': { '$gte': "58569104840549217479400131697037017088" }
});

**Reply by alexbevi** [reply]
```
db.getSiblingDB("my_db").getCollection("mongo_countries").find({
'range_from': { '$lte': "58569104840549217479400131697037017089" },
'range_to': { '$gte': "58569104840549217479400131697037017088" }
});
```

The `mongosh` query you've shared (above) represents the code variants as strings, but in Mongoid you're still trying to represent them as numbers (that are exceeding Ruby's `LLONG_MAX` limits).

Are you able to just use `ip_code = "58569104840549217479400131697037017089"`?

**Reply by oRIOnTx** [reply]
same error 
![image](https://github.com/mongodb/mongoid/assets/40061096/8382f2bc-f979-4cb9-9cbb-3253a7df375b)

**Reply by oRIOnTx** [reply]
i'm already try different type of query, but nothing helped

**Reply by alexbevi** [reply]
Can you share a self-contained reproduction that includes your `Mongo::Proxy` model? I want to make sure we're looking at the right thing.

**Reply by oRIOnTx** [reply]
`# frozen_string_literal: true

module Mongo
  class Proxy
    include Mongoid::Document

    field :range_from, type: Bignum
    field :range_to, type: Bignum
    field :code, type: String
    field :country, type: String
  end
end
`
my code next 

`
ip_code = "58569104840549217479400131697037017089"
::Mongo::Proxy.where(:range_from.lte => ip_code.to_s).and(:range_to.gte => ip_code.to_s).first
`
![image](https://github.com/mongodb/mongoid/assets/40061096/aff54d4f-283e-455c-bbb4-f9f08eb94c18)
![image](https://github.com/mongodb/mongoid/assets/40061096/ff111e79-b6ab-4bc9-b766-c049a4b02d42)

**Reply by alexbevi** [reply]
@oRIOnTx if `range_from` and `range_to` will exceed `LLONG_MAX` why not store them as `String` instead?
